torch/utils/_pytree.py

```diff
@@ -255,14 +255,26 @@ def register_pytree_node(
         _cxx_pytree_pending_imports.append((args, kwargs))
 
 
-def register_dataclass(cls: type[Any]) -> None:
-    """Registers a ``dataclasses.dataclass`` type as a pytree node.
+def register_dataclass(
+    cls: type[Any],
+    data_fields: Optional[list[str]] = None,
+    *,
```

Comment:
Can you make data_fields kwarg-only? This lets us add a meta_fields arg in the future without worrying about the position